src/App.js

```diff
@@ -26,19 +26,26 @@ function App() {
   }
 
   function handleRemoveItem(id) {
-    setShoppingCart(
-      shoppingCart.map((item) => {
-        // TODO: Remove Item from Cart if Quantity is 0
-
-        // if (item.id === id && item.quantity === 1) {
-        //   setShoppingCart(shoppingCart.filter((item) => item.id !== id));
-        // }
-        if (item.id === id) {
-          return { ...item, quantity: item.quantity - 1 };
-        }
-        return item;
-      })
-    );
+    shoppingCart.map((item) => {
+      if (item.id === id && item.quantity === 1) {
+        setShoppingCart(shoppingCart.filter((item) => item.id !== id));
+      }
+      return null;
+    });
+
+    shoppingCart.map((item) => {
+      if (item.id === id && item.quantity !== 1) {
```

Comment:
Same here, see comment above

Reply:
if you set it to !==0 you can get a "0x item" in the cart

Reply:
Okay I see. But and don't get why...